lib/kb_SetUtilities/kb_SetUtilitiesImpl.py

```diff
@@ -146,7 +170,448 @@ def get_obj_info_list_from_ws_name (self, ws_name, obj_type, obj_type_desc):
+                    raise ValueError ("assembly for genome "+genome_id+" missing all necessary fields for "+attr)
+                sorted_contig_lengths = sorted(contig_lengths, key=int, reverse=True)
+            else:
+                genome_id = _get_genome_id (genome_data, genome_tag)
+                raise ValueError ("genome "+genome_id+" missing all necessary fields for "+attr)
+            val = GC
+            
+        # CDS count
+        elif attr == 'CDS_count':
+            val = '-'
+            if len(genome_data.get('cdss',[])) > 0:
+                val = len(genome_data['cdss'])
+
+        # tRNA_count
+        elif attr == 'tRNA_count':
+            val = 0
+            #tRNAs_seen = dict()  # function string doesn't always include anti-codon
+            if 'non_coding_features' in genome_data:
+                for f in genome_data['non_coding_features']:
+                    val += _check_feature_func (f, ['tRNA'])
+                #val += len(tRNAs_seen.keys())
+
+        # 5S_rRNA_count
+        elif attr == '5S_rRNA_count':
+            val = 0
+            if 'non_coding_features' in genome_data:
+                for f in genome_data['non_coding_features']:
+                    val += _check_feature_func (f, ['5S rRNA',
+                                                    '5S ribosomal RNA'])
+
+        # 16S_rRNA_count
+        elif attr == '16S_rRNA_count':
+            val = 0
+            if 'non_coding_features' in genome_data:
+                for f in genome_data['non_coding_features']:
+                    val += _check_feature_func (f, ['16S rRNA',
+                                                    '16S ribosomal RNA',
+                                                    'SSU rRNA',
+                                                    'SSU ribosomal RNA'])
+
+        # 23S_rRNA_count
+        elif attr == '23S_rRNA_count':
+            val = 0
+            if 'non_coding_features' in genome_data:
+                for f in genome_data['non_coding_features']:
+                    val += _check_feature_func (f, ['23S rRNA',
+                                                    '23S ribosomal RNA',
+                                                    'LSU rRNA',
+                                                    'LSU ribosomal RNA'])
+
+        # CRISPR_array_count
+        elif attr == 'CRISPR_array_count':
+            val = 0
+            if 'non_coding_features' in genome_data:
+                for f in genome_data['non_coding_features']:
+                    val += _check_feature_func (f, ['CRISPR region'])
+
+        else:
+            raise ValueError ('wrong attr type for get_genome_attribute()')
+            
+        return str(val)
+
+    # get_hmmer_fams
+    def get_hmmer_fams (self, collection):
+        categories = []
+        cat_ids = dict()
+        fam_ids = dict()
+        fam_disp = dict()
+
+        if collection == 'MT_Bioelement':
+
+            categories = [ 'Nfix',
```

Comment:
I wonder if having these things in some yaml file or some other data file that could be read in.  Again, I'm not saying you have to do this but it may make it easier to maintain.

Reply:
that would be better.  kind of was just trying to get it done and moved on to next thing.  Deserves some TLC.  Next iteration!